templates/fine-tune-function-calling/fc_utils/eval_utils.py

```diff
@@ -0,0 +1,163 @@
+"""
+Evaluation utilities
+"""
+
+from typing import Union, Tuple, List
+from tqdm import tqdm
+from enum import Enum
+import ray.data
+
+from fc_utils.response_parsers import (
+    ERROR_OUTPUT,
+    AnyscaleResponseParser,
+    OpenAIResponseParser,
+    INCORRECT_FORMAT
+)
+
+
+class PossibleMistakes(Enum):
+    UNWANTED_FUNCTION_CALL = "Unwanted Function Call"
+    NO_FUNCTION_CALL = "No Function Call"
+    INCORRECT_FORMAT = "Incorrect Function Format"
+    INCORRECT_NUMBER_OF_FUNCTION_CALLS = "Incorrect Number of Function Calls"
+    WRONG_FUNCTION_NAME = "Wrong Function Name"
+    WRONG_ARGUMENT_VALUE = "Wrong Argument Value"
+    MISSING_ARGUMENT = "Missing Argument"
+
+    def values():
+        return [item.value for item in PossibleMistakes]
+
+def is_match(response: dict, ground_truth: dict) -> Tuple[bool, str]:
+    """
+    Checks if the response matches the ground truth. Returns a boolean and a string indicating the mistake type if any.
+    """
+    if ground_truth["tool_calls"] is None:
```

Comment:
comment each part so that the reader of the code can easily follow, eg:

```
# If ground truth is a normal response, so should be the actual response
```

Also rewrite for better readablity:

```
is_match = True
reason = ""


if x:
    is_match = False
    reason = Foo
elif y:
    is_match = False
    reason = Bar
...

return is_match, reason
```